internal/static/embed/install-kubeadm.sh

```diff
@@ -0,0 +1,224 @@
+
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+  SystemdCgroup = true
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+  config_path = "/etc/containerd/certs.d"
+'
+
+CONTAINERD_UNPRIVILEGED_CONFIG='
+version = 2
+
+[plugins."io.containerd.grpc.v1.cri"]
+  stream_server_address = "127.0.0.1"
+  stream_server_port = "10010"
+  enable_unprivileged_ports = true
+  enable_unprivileged_icmp = true
+  device_ownership_from_security_context = false
+  sandbox_image = "registry.k8s.io/pause:3.10"
+  enable_selinux = false
+
+  ## unprivileged
+  disable_apparmor = true
+  disable_hugetlb_controller = true
+  restrict_oom_score_adj = true
+
+[plugins."io.containerd.grpc.v1.cri".containerd]
+  snapshotter = "overlayfs"
+  disable_snapshot_annotations = true
+
+[plugins."io.containerd.grpc.v1.cri".cni]
+  bin_dir = "/opt/cni/bin"
+  conf_dir = "/etc/cni/net.d"
+
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+  runtime_type = "io.containerd.runc.v2"
+
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+  ## unprivileged
+  SystemdCgroup = false
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+  config_path = "/etc/containerd/certs.d"
+'
+
+CONTAINERD_SERVICE='
+[Unit]
+Description=containerd container runtime
+Documentation=https://containerd.io
+After=network.target local-fs.target
+
+[Service]
+#uncomment to enable the experimental sbservice (sandboxed) version of containerd/cri integration
+#Environment="ENABLE_CRI_SANDBOXES=sandboxed"
+ExecStartPre=-/sbin/modprobe overlay
+ExecStart=/usr/bin/containerd
+
+Type=notify
+Delegate=yes
+KillMode=process
+Restart=always
+RestartSec=5
+# Having non-zero Limit*s causes performance problems due to accounting overhead
+# in the kernel. We recommend using cgroups to do container-local accounting.
+LimitNPROC=infinity
+LimitCORE=infinity
+LimitNOFILE=infinity
+# Comment TasksMax if your systemd version does not supports it.
+# Only systemd 226 and above support this version.
+TasksMax=infinity
+OOMScoreAdjust=-999
+
+[Install]
+WantedBy=multi-user.target
+'
+
+CONTAINERD_CONFIGURE_UNPRIVILEGED_MODE='#!/bin/bash -xeu
+
+set -xeu
+
+ln -sf config.unprivileged.toml /etc/containerd/config.toml
+systemctl restart containerd
+'
+
+# infer ARCH
+ARCH="$(uname -m)"
+if uname -m | grep -q x86_64; then ARCH=amd64; fi
+if uname -m | grep -q aarch64; then ARCH=arm64; fi
+
+# sysctl
+echo net.ipv4.ip_forward=1 | tee /etc/sysctl.d/99-clusterapi.conf
+echo fs.inotify.max_user_instances=8192 | tee -a /etc/sysctl.d/99-clusterapi.conf
+echo fs.inotify.max_user_watches=524288 | tee -a /etc/sysctl.d/99-clusterapi.conf
+sysctl --system
+
+# kernel
+if ! systemd-detect-virt --container --quiet 2>/dev/null; then
+  modprobe br_netfilter
+  echo br_netfilter | tee /etc/modules-load.d/br_netfilter.conf
+fi
+
+# apt install requirements
+apt update
+apt install curl iptables ethtool --no-install-recommends --yes
+if [ "$KUBERNETES_VERSION" "<" "v1.32" ]; then
```

Comment:
Not sure if this is a bashism or not, but I'd use `-lt` rather than `"<"`

Reply:
the reason is pretty much: 

```bash
$ cat a.sh 
#!/bin/sh -xeu

VAR="v1.32.1"

if [ "$VAR" "<" "v1.33" ]; then echo '<' works; fi
if [ "$VAR" -lt "v1.33" ]; then echo -lt works; fi

$ ./a.sh 
+ VAR=v1.32.1
+ [ v1.32.1 < v1.33 ]
+ echo < works
< works
+ [ v1.32.1 -lt v1.33 ]
./a.sh: 6: [: Illegal number: v1.32.1
```

and it seems to work with sh as well, so can drop the bash